### DISCUSSION TITLE
How to separate a cylinder and a sphere in the same [mesh] without touching each other?

### DISCUSSION THREAD

**Discussion by Strawberrybrother** [idaholab/moose#24336] [Q&A General] [answered]
I've looked at some examples of squares or cubes, which can be separated by parameters like xmin, xmax, ymax, ymin, zmin, zmax, but how can we separate a sphere and a cylinder?Thank you for your help and answers.

**Comment by GiudGiud** [comment, marked as answer]
Hello

you can make those meshes with mesh generators then use a combiner generator to have them in the same simulation without touching. 

Unless you use interface kernels or dedicated user objects, they will not influence each other

Guillaume

**Reply by GiudGiud** [reply]
Hello

You can translate any of those two meshes using the TransformGenerator
https://mooseframework.inl.gov/docs/PRs/24293/site/source/meshgenerators/TransformGenerator.html
Guillaume

**Reply by Strawberrybrother** [reply]
Hello Guillaume,
Thank you for your help. My problem has been solved. I have another question: if I want the ball to fall vertically onto the cylinder, do I need to add [Kernels] or [BCs] or other parts?
Best regards,
Strawberrybrother

**Reply by GiudGiud** [reply]
so you want the mesh to move?

**Reply by Strawberrybrother** [reply]
Yes,I want the ball to impact this cylinder. The bottom of the cylinder is fixed . I want to observe the plastic deformation of the ball

**Reply by GiudGiud** [reply]
You should look for examples in the repository, this will be hard to build from scratch.
I think the contact module will have some

**Reply by Strawberrybrother** [reply]
Thank you very much
